source/common/common/mem_block_builder.h

```diff
@@ -26,7 +26,13 @@ template <class T> class MemBlockBuilder {
 public:
   // Constructs a MemBlockBuilder allowing for 'capacity' instances of T.
   explicit MemBlockBuilder(uint64_t capacity)
-      : data_(std::make_unique<T[]>(capacity)), write_span_(data_.get(), capacity) {}
+      : owned_data_(std::make_unique<T[]>(capacity)), write_span_(owned_data_.get(), capacity) {
+    data_ = owned_data_.get();
+  };
+  explicit MemBlockBuilder(T* data, uint64_t capacity)
```

Comment:
looks like we don't need the changes to this file now?

Reply:
Still, would it be better to leave that to another PR? Or would you prefer me to strip this off (although it does not affect the current usage of MemBlockBuilder)?

Reply:
I think if we don't need this change in this PR, let's save it for a later one. You can capture the partial diff in a file with `git diff source/common/common/mem_block_builder.h >~/diffs/mem_block_new_ctor.diff` and apply it in a future PR with `patch -p1 <~/diffs/mem_block_new_ctor.diff`.

People who are more knowledgable than me about `git` may have a better way to do this :)